shell/browser/electron_download_manager_delegate.cc

```diff
@@ -130,6 +245,29 @@ void ElectronDownloadManagerDelegate::OnDownloadPathGenerated(
     const bool offscreen = !web_preferences || web_preferences->IsOffscreen();
     settings.force_detached = offscreen;
 
+#if BUILDFLAG(IS_WIN)
+    auto extension = settings.default_path.FinalExtension();
+    if (!extension.empty() && settings.filters.empty()) {
+      extension.erase(extension.begin());
+
+      std::u16string ext(extension.begin(), extension.end());
+      const std::vector<std::u16string> file_ext{u"." + ext};
+      std::vector<ui::FileFilterSpec> filter_spec =
+          FormatFilterForExtensions(file_ext, {u""}, true, true);
+
+      std::string extension_spec_conv(filter_spec[0].extension_spec.begin(),
```

Comment:
And `base::UTF16ToUTF8` to convert `u16string` to `string`.

For unicode strings one character can be stored in multiple bytes, so for the same text `std::string` and `std::u16string` can have different lengths.